src/Security/Authentication/Cookies/samples/CookieSessionSample/Startup.cs

```diff
@@ -14,12 +14,14 @@ public class Startup
     {
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddSingleton<MemoryCacheTicketStore>();
+
             // This can be removed after https://github.com/aspnet/IISIntegration/issues/371
             services.AddAuthentication(options =>
             {
                 options.DefaultAuthenticateScheme = CookieAuthenticationDefaults.AuthenticationScheme;
                 options.DefaultChallengeScheme = CookieAuthenticationDefaults.AuthenticationScheme;
-            }).AddCookie(o => o.SessionStore = new MemoryCacheTicketStore());
+            }).AddCookie((o, services) => o.SessionStore = services.GetService<MemoryCacheTicketStore>());
```

Comment:
I hate to second guess ourselves at this point, but what if this was `AddCookie<TService>(Action<CookieAuthOptions, TService))` so the above code could read like `AddCookie<MemoryCacheTicketStore>((o, store) => o.SessionStore = store))`. It's more elegant for the single service case (most common?), and then for multi-service you can inject IServiceProvider. Options already supports this, we'd only be changing our new overloads here.
@Kahbazi @anurse @HaoK thoughts?

Reply:
Yeah that has the benefit of looking just like the options builder apis, so its more consistent, I like that much better too

Reply:
Ok I will change the api. My only concern is users may not know about `IServiceProvider` and expect to pass two or more services as generic like Options. There should be a good comment for this api. So what comment should I add for `Action<CookieAuthOptions, TService)` parameter?

Reply:
TService: A service resolved from the IServiceProvider for use when configuring this authentication provider. If you need multiple services then specify IServiceProvider and resolve them directly.